src/main/java/mServer/crawler/sender/br/BrFilmDeserializer.java

```diff
@@ -0,0 +1,393 @@
+
+        return buildFilm(detailClip, viewer.get());
+
+
+      } else {
+        printMissingDetails(JSON_ELEMENT_VIEWER);
+      }
+    } catch (final UnsupportedOperationException unsupportedOperationException) {
+      // This will happen when a element is JsonNull.
+      LOG.error("BR: A needed JSON element is JsonNull.", unsupportedOperationException);
+      FilmeSuchen.listeSenderLaufen.inc(crawler.getSendername(), RunSender.Count.FEHLER);
+    }
+
+    return Optional.empty();
+  }
+
+  private String getDescriptions(final JsonObject aDetailClip) {
+    String description = "";
+    
+    if (aDetailClip.has(JSON_ELEMENT_DESCRIPTION)
+        && !aDetailClip.get(JSON_ELEMENT_DESCRIPTION).isJsonNull()) {
+      description = aDetailClip.get(JSON_ELEMENT_DESCRIPTION).getAsString();
+    } else if (aDetailClip.has(JSON_ELEMENT_SHORT_DESCRIPTION)
+        && !aDetailClip.get(JSON_ELEMENT_SHORT_DESCRIPTION).isJsonNull()) {
+      description = aDetailClip.get(JSON_ELEMENT_SHORT_DESCRIPTION).getAsString();
+    }
+    
+    return description;
+  }
+
+  private Map<Resolution, String> getUrls(final JsonObject viewer) {
+    Map<Resolution, String> urlMap = new HashMap<>();
+    
+    final Set<BrUrlDTO> urls = edgesToUrls(viewer);
+      if (!urls.isEmpty()) {
+        // TODO ist das wirklich sinnvoll? 
+        // TODO bei drei guten Qualitäten ist Dateigröße für die schlechte immer noch ziemlich 
+        // Sorts the urls by width descending, then it limits the amount to three to get the three
+        // best.
+        final List<BrUrlDTO> bestUrls =
+            urls.stream().sorted(Comparator.comparingInt(BrUrlDTO::getWidth).reversed()).limit(3)
+                .collect(Collectors.toList());
+        
+        for (int id = 0; id < bestUrls.size(); id++) {
+          final Resolution resolution = Resolution.getResolutionFromArdAudioVideoOrdinalsByProfileName(bestUrls.get(id).getVideoProfile());
+          final String url = bestUrls.get(id).getUrl();
+          
+          if (url != null && !url.isEmpty()) {
+            if (!urlMap.containsKey(resolution)) {
+              urlMap.put(resolution, url);
+            }
+          }
+        }
+      }
+    return urlMap;
+
+  }
+
+  private Optional<DatenFilm> buildFilm(final Optional<JsonObject> detailClip, final JsonObject viewer) {
+    final Optional<DatenFilm> newFilm;
+    if (detailClip.isPresent()) {
+      String description = getDescriptions(detailClip.get());
+      Map<Resolution, String> urls = getUrls(viewer);
+      
+      if(urls.containsKey(Resolution.NORMAL) && MediathekReader.urlExists(urls.get(Resolution.NORMAL))) {
+        Optional<String> subTitle = getSubtitleUrl(viewer);
+        newFilm = createFilm(detailClip.get(), description, subTitle, urls);
+        return newFilm;
+      }
+    } else {
+      printMissingDetails(JSON_ELEMENT_DETAIL_CLIP);
+    }
+    return Optional.empty();
+  }
+
+  private Optional<String> getSubtitleUrl(JsonObject viewer) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)